### PR TITLE
add kubectl-smart

### PR DESCRIPTION
**Pull request by FingerLiu**
Signed-off-by: FingerLiu <liupeng.dalian@gmail.com>
https://github.com/FingerLiu/kubectl-smart
A kubectl plugin that makes kubectl smart with name search.Type less letters!Save more life!
Added grep support for most kubectl sub commands.

> Installing plugin: smart
Installed plugin: smart
\
 | Use this plugin:
 | 	kubectl smart
 | Documentation:
 | 	https://github.com/FingerLiu/kubectl-smart
/
Make kubectl smart with name search.Type less letters!Save more life!
USAGE:
  smart sub_command [options...] [name_pattern...]
SUB_COMMAND:
  gp                        : shortcut for get pod
  l,logs                    : shortcut for logs
  e,exec                    : shortcut for exec
  edp                       : shortcut for edit pod
  ed,edit                   : shortcut for edit
  dp                        : shortcut for delete pod
  g                         : shortcut for get
  h,help                    : show this message
OPTIONS:
  -A                        : all-namespaces
  -n                        : namespace
  -w,--wide                 : TODO get with wide output
  -f,--follow               : follow log output
  --tail                    : tail logs
  -t                        : --tty=true: Stdin is a TTY
  -i                        : --stdin=true: Pass stdin to the container
  -c                        : TODO container name.
  -e                        : use exact match rather than grep. Only use this when you want to disable grep when grep return multi items.
Examples:
```
  # get pod with name contains my in namespace her-namespace
  # (kubectl get pod -n her-namespace-a |grep my)
  kubectl smart gp -n her.*a my

  # get log for pod with name my
  # (kubectl logs --tail 100 -f $(kubectl get pods | awk '/my/ {print $1;exit}'))
  kubectl smart l my

  # exec into pod
  # kubectl exec -ti my-pod-i3jx bash
  kubectl smart e my bash

  # get deploy with name contains my
  # (kubectl get deploy | grep my)
  kubectl smart g deploy my
```

### CODE REVIEW

File: plugins/pod-lens.yaml [hunk 1]
```diff
@@ -3,14 +3,14 @@ kind: Plugin
 metadata:
   name: pod-lens
 spec:
-  version: "v0.1.2"
+  version: "v0.1.3"
```
Comment (ahmetb) [inline plugins/pod-lens.yaml:6]: Revert file

**Comment by ahmetb** [comment]
The plugin is not named "s", but the examples are?

I'm inclined to think a plugin here kind of defeats the purpose. It might be the reason why people like shell aliases and https://github.com/ahmetb/kubectl-aliases is a somewhat popular project.

**Comment by FingerLiu** [comment]
@ahmetb Thanks for reviewing.
 Reveted the file and renamed s to smart in example and usage.
>I'm inclined to think a plugin here kind of defeats the purpose. 
The main purpose of this plugin is pattern search in namespace and resource name,
which can't be done in bash alias or kubectl-aliases.

**Comment by corneliusweig** [comment]
There is a big problem with pattern matching: it is unpredictable. So I'm thinking that this plugin is better suited for being distributed via a custom plugin index ([1](https://krew.sigs.k8s.io/docs/user-guide/custom-indexes/), [2](https://krew.sigs.k8s.io/docs/developer-guide/custom-indexes/)).

You can also consider submitting to another index such as https://github.com/ishantanu/awesome-kubectl-plugins#use-awesome-kubectl-plugins-as-a-custom-index-via-krew.

**Pull request closed without merging**